src/ostree/ot-builtin-create-usb.c

```diff
@@ -0,0 +1,275 @@
+
+  if (!glnx_fstat (ostree_repo_get_dfd (dest_repo), &dest_repo_stbuf, error))
+    return FALSE;
+
+  if (dest_repo_stbuf.st_dev != mount_root_stbuf.st_dev)
+    {
+      ot_util_usage_error (context, "--destination-repo must be a descendent of MOUNT-PATH", error);
+      return FALSE;
+    }
+
+  if (ostree_repo_equal (src_repo, dest_repo))
+    {
+      ot_util_usage_error (context, "--destination-repo must not be the source repository", error);
+      return FALSE;
+    }
+
+  if (!ostree_ensure_repo_writable (dest_repo, error))
+    return FALSE;
+
+  if (opt_disable_fsync)
+    ostree_repo_set_disable_fsync (dest_repo, TRUE);
+
+  /* Copy across all of the collection–refs to the destination repo. */
+  GVariantBuilder refs_builder;
+  g_variant_builder_init (&refs_builder, G_VARIANT_TYPE ("a(sss)"));
+
+  for (gsize i = 0; i < refs->len; i++)
+    {
+      const OstreeCollectionRef *ref = g_ptr_array_index (refs, i);
+
+      g_variant_builder_add (&refs_builder, "(sss)",
+                             ref->collection_id, ref->ref_name, "");
+    }
+
+  {
+    GVariantBuilder builder;
+    g_autoptr(GVariant) opts = NULL;
+    OstreeRepoPullFlags flags = OSTREE_REPO_PULL_FLAGS_MIRROR;
+
+    glnx_console_lock (&console);
+
+    if (console.is_tty)
+      progress = ostree_async_progress_new_and_connect (ostree_repo_pull_default_console_progress_changed, &console);
+
+    g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+
+    g_variant_builder_add (&builder, "{s@v}", "collection-refs",
+                           g_variant_new_variant (g_variant_builder_end (&refs_builder)));
+    g_variant_builder_add (&builder, "{s@v}", "flags",
+                           g_variant_new_variant (g_variant_new_int32 (flags)));
+    g_variant_builder_add (&builder, "{s@v}", "depth",
+                           g_variant_new_variant (g_variant_new_int32 (0)));
+    opts = g_variant_ref_sink (g_variant_builder_end (&builder));
+
+    g_autofree char *src_repo_uri = g_file_get_uri (ostree_repo_get_path (src_repo));
+
+    if (!ostree_repo_pull_with_options (dest_repo, src_repo_uri,
+                                        opts,
+                                        progress,
+                                        cancellable, error))
+      {
+        ostree_repo_abort_transaction (dest_repo, cancellable, NULL);
+        return FALSE;
+      }
+
+    if (progress != NULL)
+      ostree_async_progress_finish (progress);
+  }
+
+  /* Ensure a summary file is present to make it easier to look up commit checksums. */
+  /* FIXME: It should be possible to work without this, but find_remotes_cb() in
+   * ostree-repo-pull.c currently assumes a summary file (signed or unsigned) is
+   * present. */
+  struct stat stbuf;
+  if (!glnx_fstatat_allow_noent (ostree_repo_get_dfd (dest_repo), "summary", &stbuf, 0, error))
+    return FALSE;
+  if (errno == ENOENT &&
+      !ostree_repo_regenerate_summary (dest_repo, NULL, cancellable, error))
```

Comment:
Need to regenerate the summary always after doing a pull, right?

Reply:
Not really; only when publishing the `summary`. Although the end result is equivalent.